File: backend/src/main/java/com/testflowai/entity/Execution.java
package com.testflowai.entity;

import lombok.Data;
import java.io.Serializable;
import java.util.Date;

/**
 * 执行记录实体类
 * 对应数据库表：t_execution
 * @author TestFlowAI
 */
@Data
public class Execution implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 执行 ID
     */
    private String executionId;

    /**
     * 测试流 ID
     */
    private String testId;

    /**
     * 执行模式 (manual/auto/scheduled)
     */
    private String mode;

    /**
     * 执行状态 (pending/running/passed/failed/stopped)
     */
    private String status;

    /**
     * 开始时间
     */
    private Date startTime;

    /**
     * 结束时间
     */
    private Date endTime;

    /**
     * 总步骤数
     */
    private Integer totalSteps;

    /**
     * 通过步骤数
     */
    private Integer passedSteps;

    /**
     * 失败步骤数
     */
    private Integer failedSteps;

    /**
     * 执行输入 JSON
     */
    private String input;

    /**
     * 执行输出 JSON
     */
    private String output;

    /**
     * 步骤结果 JSON
     */
    private String stepResults;

    /**
     * 截图路径 JSON
     */
    private String screenshots;

    /**
     * 循环上下文 JSON
     */
    private String loopContext;

    /**
     * 创建时间
     */
    private Date createdAt;

    /**
     * 更新时间
     */
    private Date updatedAt;

    /**
     * 删除时间
     */
    private Date deletedAt;

    /**
     * 逻辑删除：0-未删除，1-已删除
     */
    private Integer deleted;

    /**
     * 创建者
     */
    private String createdBy;

    /**
     * 更新者
     */
    private String updatedBy;

    /**
     * 删除者
     */
    private String deletedBy;
}

File: frontend/src/types/execution.ts
/**
 * 执行记录类型定义
 */
export interface Execution {
  executionId: string
  testId: string
  mode: string
  status: string
  startTime: string | null
  endTime: string | null
  totalSteps: number
  passedSteps: number
  failedSteps: number
  input?: string
  output?: string
  stepResults?: string
  screenshots?: string
  loopContext?: string
  createdAt: string | null
  updatedAt: string | null
  deletedAt?: string | null
  deleted?: number
  createdBy?: string
  updatedBy?: string
  deletedBy?: string
}

/**
 * 步骤结果类型定义
 */
export interface StepResult {
  resultId: string
  executionId: string
  stepId: number
  status: string
  startTime: string | null
  endTime: string | null
  errorMessage?: string
  screenshot?: string
  log?: string
  createdAt: string | null
  updatedAt: string | null
  deletedAt?: string | null
  deleted?: number
  createdBy?: string
  updatedBy?: string
  deletedBy?: string
}

/**
 * 测试报告类型定义
 */
export interface Report {
  reportId: string
  executionId: string
  testId: string
  generatedAt: string
  summary: string
  details: string
  comparisons?: string
  format: string
  filePath?: string
  createdAt: string | null
  updatedAt: string | null
  deletedAt?: string | null
  deleted?: number
  createdBy?: string
  updatedBy?: string
  deletedBy?: string
}

/**
 * 报告数据结构
 */
export interface ReportData {
  summary: {
    executionId: string
    testId: string
    status: string
    mode: string
    startTime: string
    endTime: string
    duration: string
    totalSteps: number
    passedSteps: number
    failedSteps: number
    passRate: string
  }
  details: {
    testInfo: {
      title: string
      version: string
      appUrl: string
    }
    executionInfo: {
      executionId: string
      mode: string
      status: string
    }
    stepDetails: Array<{
      stepId: number
      status: string
      startTime: string
      endTime: string
      duration: string
      errorMessage?: string
      screenshot?: string
      log?: string
    }>
  }
}
